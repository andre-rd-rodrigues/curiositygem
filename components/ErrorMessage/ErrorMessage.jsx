import AppButton from "components/AppButton/AppButton";
import React from "react";
import styles from "./errormessage.module.scss";

const ErrorMessage = () => {
  return (
    <div className={styles.container}>
      <h2>Upsss...</h2>
      <p>Something went really wrong</p>
      <AppButton
        label="Refresh page"
        onClick={() => window.location.reload()}
      />
    </div>
  );
};

export default ErrorMessage;
