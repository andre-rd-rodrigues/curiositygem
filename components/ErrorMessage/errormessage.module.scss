@import "assets/styles/mixins";

.container {
  text-align: center;
  h2 {
    margin-bottom: 20px;
  }
}
