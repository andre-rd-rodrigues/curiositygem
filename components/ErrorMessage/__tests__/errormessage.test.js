import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ErrorMessage from "../ErrorMessage";

describe("Error message", () => {
  it("renders component unchanged", () => {
    const { container } = render(<ErrorMessage />);

    expect(container).toMatchSnapshot();
  });
});
