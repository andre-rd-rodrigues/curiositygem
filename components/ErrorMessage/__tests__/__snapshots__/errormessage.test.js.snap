// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error message renders component unchanged 1`] = `
<div>
  <div
    class="container "
  >
    <div
      role="button"
      style="cursor: pointer; display: inline;"
      tabindex="0"
    >
      <svg
        class="feather feather-alert-circle "
        color="primary"
        fill="none"
        height="80"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="stroke-width: 1;"
        viewBox="0 0 24 24"
        width="80"
      >
        <g>
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="16"
            y2="16"
          />
        </g>
      </svg>
    </div>
    <h2>
      Ohhh no...
    </h2>
    <p>
      Something went wrong while getting the articles
    </p>
    <button
      class="container"
    >
      Refresh page
    </button>
  </div>
</div>
`;
